API provides user with a list of cocktails

johanbounce opened the pull request:
# [PT Story](https://www.pivotaltracker.com/story/show/171699845)
```
As a user
In order to get an overview of possible cocktails
I would like to get a list of cocktails based on my search
```
## Changes proposed in this pull request:
* Adds test for index action in CocktailController 
* Adds index action in CocktailController
* Stubbing out request for cocktail list
## What I have learned working on this feature:
* Working with the gem of WebMock
* Stubbing out request of http request
## Packages/Gems added
- WebMock
- RestClient

emtalen committed:
generating cocktails controller and creates spec for the controller

johanbounce committed:
Creating index method for cocktailscontroller and adding request specs

emtalen committed:
adding webmock gem and inserts the stub from the error message into rails_helper

johanbounce committed:
stubbed out get request for cocktails index. added fixture file and configured stub in rails helper

emtalen committed:
Changing the param in the spec to Lime_juice and removing the headers from stub_requests

johanbounce committed:
Finalizing searching for cocktail list

robin-lillqvist commented on app/controllers/api/cocktails_controller.rb:
```diff
@@ -0,0 +1,18 @@
+      {
+        params: {
+          i: params[:q],
+        },
```
are these commas needed? not a big thing, but it looks like we're expecting multiple of these?

johanbounce replied:
Done

robin-lillqvist reviewed (commented):
Looks good, just one question in the code.
If you could check if it still works without the commas. If not then it's gucci

johanbounce committed:
Deleted unneeded commas in cocktails controll

kianaditya commented on spec/requests/api/user_can_see_list_of_cocktails_spec.rb:
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+RSpec.describe Api::CocktailsController, type: :request do
```
You need to test sad path

johanbounce replied:
Fixed

kianaditya commented on app/controllers/api/cocktails_controller.rb:
```diff
@@ -0,0 +1,18 @@
+class Api::CocktailsController < ApplicationController
+  def index
+    response = RestClient.get(
```
You should check incoming request has query params before making this call

johanbounce replied:
Fixed

johanbounce committed:
Refactoring AVAILABLE_INGREDIENTS constant to application controller,
checking incomming query params before making request to api
writing sad path test for user can see list of cocktails

johanbounce committed:
Refactoring AVAILABLE_INGREDIENTS constant to application controller,
checking incomming query params before making request to api
writing sad path test for user can see list of cocktails

kianaditya commented on app/controllers/api/cocktails_controller.rb:
```diff
@@ -0,0 +1,22 @@
+    )
+    results = JSON.parse(response)
+      render json: { drinks: results["drinks"] }
+      binding.pry
```
remove `binding.pry`

kianaditya commented on app/controllers/api/cocktails_controller.rb:
```diff
@@ -0,0 +1,22 @@
+      render json: { drinks: results["drinks"] }
+      binding.pry
+    else
+    render json: { message: "You can not choose this ingredient" }, status: 422
```
formatting

kianaditya commented on app/controllers/api/cocktails_controller.rb:
```diff
@@ -0,0 +1,22 @@
+    end
+  end
+
+  def show
```
Why do you need `show` ?

johanbounce committed:
removed binding.pry, formatted code, and deleted empty show method.

kianaditya reviewed (approved):
LGTM

tochman commented on app/controllers/api/cocktails_controller.rb:
```diff
@@ -0,0 +1,18 @@
+        }
+      )
+      results = JSON.parse(response)
+        render json: { drinks: results["drinks"] }
```
Formatting, please

emtalen committed:
fixes formatting in cocktails controller
